[Style] Convert more SVG length properties to strong style types

weinig opened the pull request:
#### de7c13fec64754ecc10dd34d502c1743eaab1373
<pre>
[Style] Convert more SVG length properties to strong style types
<a href="https://bugs.webkit.org/show_bug.cgi?id=296536">https://bugs.webkit.org/show_bug.cgi?id=296536</a>

Reviewed by Darin Adler.

Converts the remaining SVG properties that contain a &lt;length&gt; to use
strong style types. These include `baseline-shift`, `cx`, `cy`, `r`,
`rx`, `ry`, `x`, `y`, `stroke-dasharray` and `stroke-dashoffset`.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/SVGRenderStyle.cpp:
* Source/WebCore/rendering/style/SVGRenderStyle.h:
* Source/WebCore/rendering/style/SVGRenderStyleDefs.cpp:
* Source/WebCore/rendering/style/SVGRenderStyleDefs.h:
* Source/WebCore/rendering/svg/RenderSVGEllipse.cpp:
* Source/WebCore/rendering/svg/RenderSVGRect.cpp:
* Source/WebCore/rendering/svg/SVGRenderSupport.cpp:
* Source/WebCore/rendering/svg/SVGRenderTreeAsText.cpp:
* Source/WebCore/rendering/svg/SVGTextLayoutEngineBaseline.cpp:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGEllipse.cpp:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGRect.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleExtractorCustom.h:
* Source/WebCore/style/StyleInterpolationFunctions.h:
* Source/WebCore/style/StyleInterpolationWrappers.h:
* Source/WebCore/style/values/svg/StyleSVGBaselineShift.cpp: Added.
* Source/WebCore/style/values/svg/StyleSVGBaselineShift.h: Added.
* Source/WebCore/style/values/svg/StyleSVGCenterCoordinateComponent.h: Added.
* Source/WebCore/style/values/svg/StyleSVGCoordinateComponent.h: Added.
* Source/WebCore/style/values/svg/StyleSVGRadius.h: Added.
* Source/WebCore/style/values/svg/StyleSVGRadiusComponent.h: Added.
* Source/WebCore/style/values/svg/StyleSVGStrokeDasharray.cpp: Added.
* Source/WebCore/style/values/svg/StyleSVGStrokeDasharray.h: Added.
* Source/WebCore/style/values/svg/StyleSVGStrokeDashoffset.cpp: Added.
* Source/WebCore/style/values/svg/StyleSVGStrokeDashoffset.h: Added.
* Source/WebCore/svg/SVGLengthContext.cpp:
* Source/WebCore/svg/SVGLengthContext.h:

Canonical link: <a href="https://commits.webkit.org/297909@main">https://commits.webkit.org/297909@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/78d993c14461c945acaf07757cb7eba34ac1fb05

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113325 "11 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33057 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23529 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119524 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/64093 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/115278 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33704 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41623 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86249 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/41343 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116272 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26927 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101937 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66577 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26192 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20060 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63278 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96307 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20136 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122750 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40395 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30147 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95103 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40780 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98146 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94849 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24207 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39987 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17805 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/36555 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40280 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/45779 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39927 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43254 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41666 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/78d993c14461c945acaf07757cb7eba34ac1fb05. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48570

weinig committed:
[Style] Convert more SVG length properties to strong style types
https://bugs.webkit.org/show_bug.cgi?id=296536

Reviewed by Darin Adler.

Converts the remaining SVG properties that contain a <length> to use
strong style types. These include `baseline-shift`, `cx`, `cy`, `r`,
`rx`, `ry`, `x`, `y`, `stroke-dasharray` and `stroke-dashoffset`.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/SVGRenderStyle.cpp:
* Source/WebCore/rendering/style/SVGRenderStyle.h:
* Source/WebCore/rendering/style/SVGRenderStyleDefs.cpp:
* Source/WebCore/rendering/style/SVGRenderStyleDefs.h:
* Source/WebCore/rendering/svg/RenderSVGEllipse.cpp:
* Source/WebCore/rendering/svg/RenderSVGRect.cpp:
* Source/WebCore/rendering/svg/SVGRenderSupport.cpp:
* Source/WebCore/rendering/svg/SVGRenderTreeAsText.cpp:
* Source/WebCore/rendering/svg/SVGTextLayoutEngineBaseline.cpp:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGEllipse.cpp:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGRect.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleExtractorCustom.h:
* Source/WebCore/style/StyleInterpolationFunctions.h:
* Source/WebCore/style/StyleInterpolationWrappers.h:
* Source/WebCore/style/values/svg/StyleSVGBaselineShift.cpp: Added.
* Source/WebCore/style/values/svg/StyleSVGBaselineShift.h: Added.
* Source/WebCore/style/values/svg/StyleSVGCenterCoordinateComponent.h: Added.
* Source/WebCore/style/values/svg/StyleSVGCoordinateComponent.h: Added.
* Source/WebCore/style/values/svg/StyleSVGRadius.h: Added.
* Source/WebCore/style/values/svg/StyleSVGRadiusComponent.h: Added.
* Source/WebCore/style/values/svg/StyleSVGStrokeDasharray.cpp: Added.
* Source/WebCore/style/values/svg/StyleSVGStrokeDasharray.h: Added.
* Source/WebCore/style/values/svg/StyleSVGStrokeDashoffset.cpp: Added.
* Source/WebCore/style/values/svg/StyleSVGStrokeDashoffset.h: Added.
* Source/WebCore/svg/SVGLengthContext.cpp:
* Source/WebCore/svg/SVGLengthContext.h:

Canonical link: https://commits.webkit.org/297909@main

webkit-commit-queue commented:
Committed 297909@main (de7c13fec647): <https://commits.webkit.org/297909@main>

Reviewed commits have been landed. Closing PR #48570 and removing active labels.
